src/player.ts

```diff
@@ -174,32 +114,51 @@ export class Player {
   cleanup = async (signal = 'SIGINT'): Promise<boolean> => {
     this.logger.debug("calling Player.cleanup, conductor: %b", this._conductor)
     if (this._conductor) {
-      await this.kill(signal)
+      await this.shutdown(signal)
       unparkPort(this._adminInterfacePort)
-      unparkPort(this._appInterfacePort)
       return true
     } else {
       unparkPort(this._adminInterfacePort)
-      unparkPort(this._appInterfacePort)
       return false
     }
   }
 
-  _setCellNicks = async () => {
-    const { cell_data } = await this._conductor!.appClient!.appInfo({ app_id: 'LEGACY' })
-    for (const [cellId, cellNick] of cell_data) {
-      this._cellIds[cellNick] = cellId
-    }
+  /**
+   * helper to create agent keys and install multiple apps for scenario initialization
+   */
+  installAgentsHapps = async (agentsHapps: InstallAgentsHapps): Promise<InstalledAgentHapps> => {
+    this._conductorGuard(`Player.installHapps`)
+    return await Promise.all(agentsHapps.map(async agentHapps => {
```

Comment:
we need to put`agentPubKey` generation back in at this level

Reply:
why?  `installAgentsHapps` assumes that each element of the array has a different agent.  So `installHapp` will do the right thing.

Reply:
when you showed me example.ts you showed me how you wanted a case where two different happs were installed with the SAME agent, and that's the whole reason for a three deep hierarchy. Otherwise we're just saying `1 agent -> 1 happs` and no happs installed this way will share the same one.

Reply:
fixed